tests/rados/test_pg_split.py

```diff
@@ -0,0 +1,87 @@
+from tests.rados.rados_test_util import create_pools, write_to_pools
+from tests.rados.stretch_cluster import wait_for_clean_pg_sets
+from utility.log import Log
+from utility.utils import method_should_succeed, should_not_be_empty
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Automates PG split and merge test scenarios.
+    1. Create replicated pool
+    2.
+    """
+    try:
+        log.info(run.__doc__)
+        config = kw["config"]
+        rhbuild = config.get("rhbuild")
+        cephadm = CephAdmin(cluster=ceph_cluster, **config)
+        rados_obj = RadosOrchestrator(node=cephadm)
+        pool_obj = PoolFunctions(node=cephadm)
+        client_node = ceph_cluster.get_nodes(role="client")[0]
+        timeout = config.get("timeout", 10800)
+
+        log.info("Running PG split merge scenarios")
+        pool = create_pools(config, rados_obj, client_node)
+        should_not_be_empty(pool, "Failed to retrieve pool details")
+        write_to_pools(config, rados_obj, client_node)
+        # Set recover threads configurations
+        if not rhbuild.startswith("6"):
+            rados_obj.change_recover_threads(config=pool, action="set")
+        method_should_succeed(wait_for_clean_pg_sets, rados_obj, timeout)
+        pg_no = rados_obj.get_pool_property(pool=pool["pool_name"], props="pg_num")
+        if not pool["pg_num"] == pg_no:
+            log.error(
+                f"Actual pg_num {pg_no} does not match with expected pg_num {pool['pg_num']}"
+            )
+            raise Exception(
+                f"Actual pg_num {pg_no} does not match with expected pg_num {pool['pg_num']}"
+            )
+        bulk = pool_obj.get_bulk_details(pool["pool_name"])
+        if bulk:
+            log.error("Expected bulk flag should be False.")
+            raise Exception("Expected bulk flag should be False.")
+        new_bulk = pool_obj.set_bulk_flag(pool["pool_name"])
+        if not new_bulk:
+            log.error("Expected bulk flag should be True.")
+            raise Exception("Expected bulk flag should be True.")
+        method_should_succeed(wait_for_clean_pg_sets, rados_obj, timeout)
+        new_pgno = rados_obj.get_pool_property(pool=pool["pool_name"], props="pg_num")
+        if not new_pgno == "512":
+            log.error(
+                f"Actual pg_num {pg_no} does not match with expected pg_num {pool['pg_num']}"
+            )
+            raise Exception(
+                f"Actual pg_num {pg_no} does not match with expected pg_num {pool['pg_num']}"
+            )
+        rm_bulk = pool_obj.rm_bulk_flag(pool["pool_name"])
+        if rm_bulk:
+            log.error("Expected bulk flag should be False.")
+            raise Exception("Expected bulk flag should be False.")
+        method_should_succeed(wait_for_clean_pg_sets, rados_obj, timeout)
+        npg_no = rados_obj.get_pool_property(pool=pool["pool_name"], props="pg_num")
+        if not pool["pg_num"] == npg_no:
```

Comment:
Again, We cannot be sure if the PG count would reach the levels, pre-bulk settings, as determined by Autoscaler based on no of objects written etc.

ONce again, we can just check if ` new_pgno ` > `npg_no` . If we can make sure that the PG count decreased, we can be sure of PG merges happening.

Reply:
made the changes